File: java/src/test/java/io/binac/leetcode/TextJustificationTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class TextJustificationTest {
    private static final TextJustification.Solution1 SOLUTION1 = new TextJustification.Solution1();

    private final String[] WORDS1 = {"This", "is", "an", "example", "of", "text", "justification."};
    private final int MAXWIDTH1 = 16;
    private final List<String> EXPECTED1 = Arrays.asList(
            "This    is    an",
            "example  of text",
            "justification.  "
    );

    private final String[] WORDS2 = {"What", "must", "be", "acknowledgment", "shall", "be"};
    private final int MAXWIDTH2 = 16;
    private final List<String> EXPECTED2 = Arrays.asList(
            "What   must   be",
            "acknowledgment  ",
            "shall be        "
    );

    private final String[] WORDS3 = {"Science", "is", "what", "we", "understand", "well", "enough", "to", "explain",
            "to", "a", "computer.", "Art", "is", "everything", "else", "we", "do"};
    private final int MAXWIDTH3 = 20;
    private final List<String> EXPECTED3 = Arrays.asList(
            "Science  is  what we",
            "understand      well",
            "enough to explain to",
            "a  computer.  Art is",
            "everything  else  we",
            "do                  "
    );

    @Test
    void test1() {
        assertEquals(EXPECTED1, SOLUTION1.fullJustify(WORDS1, MAXWIDTH1));
        assertEquals(EXPECTED2, SOLUTION1.fullJustify(WORDS2, MAXWIDTH2));
        assertEquals(EXPECTED3, SOLUTION1.fullJustify(WORDS3, MAXWIDTH3));
    }
}

File: cpp/tests/TextJustificationTest.cpp
#include "TextJustification.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(TextJustification, Solution68_1) {
  auto S68_1 = Solution68_1();

  std::vector<std::string> Words1{
      "This", "is", "an", "example", "of", "text", "justification."
  };
  const int MaxWidth1 = 16;
  const std::vector<std::string> Expected1{
      "This    is    an",
      "example  of text",
      "justification.  "
  };
  EXPECT_EQ(Expected1, S68_1.fullJustify(Words1, MaxWidth1));

  std::vector<std::string> Words2{
      "What", "must", "be", "acknowledgment", "shall", "be"
  };
  const int MaxWidth2 = 16;
  const std::vector<std::string> Expected2{
      "What   must   be",
      "acknowledgment  ",
      "shall be        "
  };
  EXPECT_EQ(Expected2, S68_1.fullJustify(Words2, MaxWidth2));

  std::vector<std::string> Words3{
      "Science", "is", "what", "we", "understand", "well", "enough", "to",
      "explain", "to", "a", "computer.", "Art", "is", "everything", "else",
      "we", "do"
  };
  const int MaxWidth3 = 20;
  const std::vector<std::string> Expected3{
      "Science  is  what we",
      "understand      well",
      "enough to explain to",
      "a  computer.  Art is",
      "everything  else  we",
      "do                  "
  };
  EXPECT_EQ(Expected3, S68_1.fullJustify(Words3, MaxWidth3));
}